pharma_game.py

```diff
@@ -0,0 +1,174 @@
+import itertools
+import json
+from typing import List
+import pygame
+import socket
+import time
+
+# Constants
+PANEL_SIZE = 16           # Size of a single panel on the cross, in pixels
+SCREEN_SIZE = 3 * PANEL_SIZE  # Width and height of the cross, in pixels
+PIXEL_SIZE = 20           # Width of each square representing an LED
+PIXEL_RADIUS_RATIO = 0.5  # Relative diameter of each LED
+FPS_2COLOR = 60           # Frame rate for 2-color mode
+FPS_8COLOR = 20           # Frame rate for 8-color mode
+GREEN_BRIGHTNESS = 220    # Brightness (0-255) of the brightest green
+COLOR_DEPTH = 3           # Number of bits in each shade of green
+GAME_TIME_LIMIT = 30      # Time limit for the game (seconds)
+
+
+class PharmaScreen:

```

Comment:
It would probably be better to import `PharmaScreen` from pharmacontroller.py than copy pasting it

Otherwise I did not go into details but it looks good to me